lib/models/offerings_wrapper.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+import 'offering_wrapper.dart';
+
+part 'offerings_wrapper.freezed.dart';
+part 'offerings_wrapper.g.dart';
+
+@freezed
+
+/// This class contains all the offerings configured in RevenueCat dashboard.
+/// For more info see https://docs.revenuecat.com/docs/entitlements
+class Offerings with _$Offerings {
+  const factory Offerings(
+    /// Map of all Offerings [Offering] objects keyed by their identifier.
+    @JsonKey(name: 'all') Map<String, Offering> all, {
+
+    /// Current offering configured in the RevenueCat dashboard.
+    @JsonKey(name: 'current', nullable: true) Offering? current,
+  }) = _Offerings;
+
+  factory Offerings.fromJson(Map<String, dynamic> json) =>
```

Comment:
this class looks much better than before, but the one missing element from it is the ability to [get an offering by name](https://github.com/RevenueCat/purchases-flutter/blob/main/lib/offerings_wrapper.dart#L20), could you add it?

Reply:
checked for other getter, my bad - fixed